src/self_encryptor.rs

```diff
@@ -1483,6 +1715,81 @@ mod tests {
         Ok(())
     }
 
+    async fn general_truncate_test(
+        old_size: u32,
+        new_size: u32,
+        num_chunks: u32,
```

Comment:
this `num_chunks` is the `expected number of chunks after truncate` ?
better have a comment explain the general purpose of this function.